pkg/mutation/assign_mutator.go

```diff
@@ -131,3 +135,110 @@ func applyToToBindings(applyTos []mutationsv1alpha1.ApplyTo) []SchemaBinding {
 	}
 	return res
 }
+
+// IsValidAssign returns an error if the given assign object is not
+// semantically valid
+func IsValidAssign(assign *mutationsv1alpha1.Assign) error {
+	path, err := parser.Parse(assign.Spec.Location)
+	if err != nil {
+		return errors.Wrapf(err, "Invalid location format")
+	}
+
+	if hasMetadataRoot(path) {
+		return fmt.Errorf("Assign can't change metadata")
+	}
+
+	err = checkKeyNotChanged(path)
+	if err != nil {
+		return err
+	}
+
+	toAssign := make(map[string]interface{})
+	err = json.Unmarshal([]byte(assign.Spec.Parameters.Assign.Raw), &toAssign)
+	if err != nil {
+		return errors.Wrapf(err, "Invalid format for parameters.assign")
```

Comment:
nit: `Wrapf` -> `Wrap`, since we arent using formatting args

This is in a few places, but I'm only leaving the one comment